wasm/encode.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package wasm
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+
+	"github.com/go-interpreter/wagon/wasm/leb128"
+)
+
+const currentVersion = 0x01
+
+func EncodeModule(w io.Writer, m *Module) error {
```

Comment:
this will need a doc-comment.

~~also, perhaps, just for the sake of symmetry, we could rename it `WriteModule`?~~
~~it would be unfortunate that we have on one hand `ReadModule` and on the other hand `EncodeModule`...~~

ok, I see you introduced `DecodeModule`. NVM.